### DISCUSSION TITLE
Time-dependent dislocation density

### DISCUSSION THREAD

**Discussion by zahidhasan83** [idaholab/moose#27306] [Q&A Modules: Phase field]
Dear MOOSE experts,
I am using DeformedGrainMaterial to accumulate dislocations in grains. I am trying to make the dislocation density (Disloc_Den) time-dependent. For that the input file looks like:
```
[Functions]
  [./Dislocden_Gradient]
    type = ParsedFunction
    expression = '5e+17 + 10*t'
  [../]
[]

[AuxVariables]
[./Dislocden]
[../]
[]

[AuxKernels]
[./Dislocden_grad]
    type = FunctionAux
    variable = Dislocden
    function = Dislocden_Gradient
    execute_on = TIMESTEP_END
  [../]
[]

[Materials]
[./deformed]
    type = DeformedGrainMaterial
    int_width = 4.0
    Burg_vec = 3.9e-10
    Disloc_Den = Dislocden
    Elas_Mod = 73e+9
    GBE = 1.64
    GBMobility = 2e-13
    grain_tracker = grain_tracker
    v = 'gr0 gr1 gr2 gr3'
    outputs = exodus
  [../]
[]
```

However, running the file it shows an error "invalid float syntax for parameter: Materials/deformed/Disloc_Den=Dislocden".
What is the problem, would you be kind to let me know?

Thanks for your support,
Zahid

**Comment by GiudGiud** [comment]
Hello

the material you are using expects a floating point number for the Disloc_Den number.
Unless this material also has a Disloc_Den_function parameter, you will need to either use a different material that lets you specify a function, or modify that material a little bit to use a function. 

That modification would be really easy. There are many examples of using functions in material properties.

Guillaume

**Reply by zahidhasan83** [reply]
Dear Dr. Guillaume,
I appreciate your reply. Since I am a beginner in MOOSE, I am not familiar with modifications of materials to use a function. Also, I am not sure where to find those examples you have mentioned. Would you be kind to direct me to the right place where those examples are stored?

I highly appreciate your time.

Thanks and best regards,
Zahid

**Reply by GiudGiud** [reply]
hello

you can see a material using a function here
https://github.com/idaholab/moose/blob/next/framework/src/materials/GenericFunctionMaterial.C

dont pay attention to the `<is_ad>` and template stuff, you do not need that.

specifically this routine for the evaluation
GenericFunctionMaterialTempl<is_ad>::computeQpFunctions()

Examples of code are in the repository. src/ for source file, include/ for headers. I recommend you follow the MOOSE tutorial to learn how to create code that runs in moose
